core/src/main/java/org/hisp/dhis/android/core/event/internal/EventQuery.kt

```diff
@@ -25,19 +25,18 @@
  *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
  *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
+package org.hisp.dhis.android.core.event.internal
 
-package org.hisp.dhis.android.core.arch.call.queries.internal;
+import org.hisp.dhis.android.core.arch.call.queries.internal.BaseQuery
+import org.hisp.dhis.android.core.trackedentity.internal.TrackerQueryCommonParams
 
-import com.google.auto.value.AutoValue;
-
-import java.util.Set;
-
-@AutoValue
-public abstract class UidsQuery extends BaseQuery {
-
-    public abstract Set<String> uids();
-
-    public static UidsQuery create(Set<String> uids) {
-        return new AutoValue_UidsQuery(1, BaseQuery.DEFAULT_PAGE_SIZE, false, uids);
-    }
-}
+internal data class EventQuery(
+    val commonParams: TrackerQueryCommonParams,
+    val orgUnit: String? = null,
+    val uids: Collection<String?> = emptyList(),
```

Comment:
Could it be `Collection<String>` (nonNull)?

Reply:
Yes, in fact this internal class is only used in its own test. Maybe it could be removed?

Reply:
Really? If so, it could be removed